### PR TITLE
feat: for tap sign in, automatically fill text box with badge number

### PR DESCRIPTION
**Pull request by mathcolo**
Asana Task: [📐 For tap sign in, automatically fill text box with badge number](https://app.asana.com/0/1200273269966439/1207900898874751/f)

Title says it all!

Checklist

- Tests:
  - `(x)` Has tests
  - `( )` Doesn't need tests
  - `( )` Tests deferred (with justification)
- Product/Design sign off:
  - `(x)` Okayed the plan for the feature (e.g. the design files, or the Asana task)
  - `( )` Reviewed the feature as implemented (e.g. on dev-green, or saw screenshots)
  - `( )` No review needed

### CODE REVIEW

File: js/test/components/operatorSignIn/attestation.test.tsx [hunk 1]
```diff
@@ -49,22 +52,39 @@ describe("Attestation", () => {
...
+        <Attestation
+          badge="123"
+          prefill={false}
+          onComplete={jest.fn()}
+          loading={false}
+          employees={EMPLOYEES}
+        />,
+      );
+      expect(view.getByRole("textbox")).toBeInTheDocument();
+    });
```
Comment (lemald) [inline js/test/components/operatorSignIn/attestation.test.tsx:67]: **nitpick:** when `prefill` is `false` can you add an additional assertion that the field starts out unfilled?

File: js/test/components/operatorSignIn/operatorSignInModal.test.tsx [hunk 2]
```diff
@@ -72,6 +72,7 @@ describe("OperatorSignInModal", () => {
     await userEvent.click(view.getByRole("button", { name: "OK" }));
 
     expect(view.getByText("Step 2 of 2")).toBeInTheDocument();
+    expect(view.getByRole("textbox")).toHaveValue(""); // Not pre-filled if manually typed
```
Comment (lemald) [inline js/test/components/operatorSignIn/operatorSignInModal.test.tsx:75]: **question:** Hmm, this isn't a change that you made, but now I'm confused: why is it going through the type workflow here? At the top of the file we mock `useNfcSupported` to return `true` and it's only overridden in one other test, with a `mockReturnValueOnce` for that matter.

File: js/test/components/operatorSignIn/attestation.test.tsx (hunk 1 above)
Comment (mathcolo) [reply js/test/components/operatorSignIn/attestation.test.tsx]: 👍 2b62b02

File: js/test/components/operatorSignIn/operatorSignInModal.test.tsx (hunk 2 above)
Comment (mathcolo) [reply js/test/components/operatorSignIn/operatorSignInModal.test.tsx:75]: Ah! The two workflows are not mutually exclusive. Even when `useNfc` is returning `reading`, you can type a badge number and proceed that way.

Comment (lemald) [reply js/test/components/operatorSignIn/operatorSignInModal.test.tsx:75]: Ohh, right, of course! Sorry for the confusion.

**Review by lemald — APPROVED** [review]
👍

File: js/test/components/operatorSignIn/operatorSignInModal.test.tsx (hunk 2 above)
Comment (mathcolo) [reply js/test/components/operatorSignIn/operatorSignInModal.test.tsx:75]: No prob 😃

**Pull request merged**